### PR TITLE
fixes 37

### LINKED ISSUE #37: Add method to return Condition for if school has user-selected field in its top 5 programs

### PR DESCRIPTION
**Pull request by jsung23**
fixes #37

### CODE REVIEW

File: CollegeMatchWebApp/src/main/java/School.java [hunk 1]
```diff
@@ -909,4 +927,85 @@ public int getId() {
...
+
+	public int getPopProg1ID() {
+		return popProg1ID;
+	}
+
+	public void setPopProg1ID(int popProg1ID) {
+		this.popProg1ID = popProg1ID;
+	}
+
+	public boolean isPopProg1IDIsNotNull() {
```
Comment (geomcmaster) [inline CollegeMatchWebApp/src/main/java/School.java]: Can you rename these along the lines of isPopProg1IDNotNull? The extra _is_ that eclipse has in the autogenerated names really bothers me haha.

Comment (jsung23) [inline CollegeMatchWebApp/src/main/java/School.java]: Lol sure

File: d [hunk 2]
```diff
@@ -0,0 +1,254 @@
+[33mcommit 4cfd71fdcf78274d63995554b1683aea970b5be1[m[33m ([m[1;36mHEAD -> [m[1;32mmaster[m[33m, [m[1;31morigin/master[m[33m, [m[1;31morigin/HEAD[m[33m)[m
```
Comment (geomcmaster) [inline d]: I don't know how you get all these weird files haha

File: CollegeMatchWebApp/src/test/java/SchoolDAOTest.java [hunk 3]
```diff
@@ -43,6 +46,33 @@ public void testGetSchools() {
...
+			schoolFields.add(school.getPopProg1ID());
+			if (school.isPopProg2IDIsNotNull() == true)
+			schoolFields.add(school.getPopProg2ID());
+			if (school.isPopProg3IDIsNotNull() == true)
+			schoolFields.add(school.getPopProg3ID());
+			if (school.isPopProg4IDIsNotNull() == true)
+			schoolFields.add(school.getPopProg4ID());
+			if (school.isPopProg5IDIsNotNull() == true)
+			schoolFields.add(school.getPopProg5ID());
+			assertTrue("School contains user-selected field of study in Top 5", !Collections.disjoint(fieldIDList, schoolFields));
```
Comment (geomcmaster) [inline CollegeMatchWebApp/src/test/java/SchoolDAOTest.java]: Did not know about the disjoint method. Cool!

File: CollegeMatchWebApp/src/main/java/SchoolDAO.java [hunk 4]
```diff
@@ -172,10 +178,12 @@ private StringBuilder buildORGroupConditionString(Condition cond, Index i) {
 	 * @return A StringBuilder for SELECT, FROM and JOIN clauses
 	 */
 	private StringBuilder selectAndJoin(byte tablesToJoin) {
+		//Added pop_prog's to baseQuery b/c I needed to grab them in my JUnit test for the containsSelectedFieldsOfStudy method
```
Comment (geomcmaster) [inline CollegeMatchWebApp/src/main/java/SchoolDAO.java]: I don't want to add stuff to the base query just for testing.

File: CollegeMatchWebApp/resources/db.properties [hunk 5]
```diff
@@ -0,0 +1,2 @@
+user=jsung
```
Comment (geomcmaster) [inline CollegeMatchWebApp/resources/db.properties]: Don't commit this file

**Comment by jsung23** [comment]
Got rid of popProgIDs in baseQuery. Got rid of popProgID setters in processResults. Changed JUnit test to be independent of popProgID getters. NOTE: I have 7,098 schools in my school table, but I don't think that's all of them, so to compare the returned list of schoolNames with my omnisciently-gathered list of schoolNames for comparison, I did containsAll instead of exact match of the 2 lists.

File: CollegeMatchWebApp/src/main/java/SchoolDAO.java (hunk 4 above)
Comment (geomcmaster) [inline CollegeMatchWebApp/src/main/java/SchoolDAO.java]: You can get rid of that comment now

**Comment by jsung23** [comment]
True... I can repopulate tables and hopefully I'll get every school. I can
write the test after that to be exact match.

**Comment by jsung23** [comment]
Made JUnit test method for containFieldsOfStudy not dependent on querying popProg IDs. Good to go.

**Comment by jsung23** [comment]
db.properties is gone now.

**Pull request merged**